openfisca_core/simulations.py

```diff
@@ -183,11 +202,15 @@ def calculate_add(self, variable_name, period, **parameters):
                 period))
 
         return sum(
-            self.calculate(variable_name, sub_period, **parameters)
+            self.calculate_(population, variable_name, sub_period, **parameters)
             for sub_period in period.get_subperiods(variable.definition_period)
             )
 
     def calculate_divide(self, variable_name, period, **parameters):
+        population = self.get_variable_population(variable_name)
+        return self.calculate_divide_(population, variable_name, period, **parameters)
+
+    def calculate_divide_(self, population, variable_name, period, **parameters):
```

Comment:
I agree, and would like to find better naming. 

With this PR's naming: 

- `calculate` is the current `calculate`. It calculates a variable for the whole simulation population. 
- `calculate_` does the same thing, but for a specific population, that may be a subset of the simulation population. It is lower-level, and should not be directly used by a user. Yet it needs to be accessible by some `population`  methods, so it can't be fully private. 

It looks like a [single `_` prefix means "protected" in Python](https://stackoverflow.com/questions/797771/python-protected-attributes), so maybe just moving the `_` as a prefix would be good enough?